presto-main/src/main/java/com/facebook/presto/operator/OperatorContext.java

```diff
@@ -213,6 +223,26 @@ public void reserveMemory(long bytes)
         }
     }
 
+    public boolean tryReserveMemory(long bytes)
+    {
+        try {
+            ListenableFuture<?> future = driverContext.reserveMemory(bytes);
+            if (!future.isDone()) {
+                return false;
```

Comment:
This is a bit strange.  Consider adding ` tryReserveMemory` to `{Driver|Pipeline|Task}Context`